snakefmt/parser/parser.py

```diff
@@ -105,42 +101,45 @@ def process_keyword(self, status):
         keyword = status.token.string
         accepts_py = True if keyword in accept_python_code else False
         new_grammar = self.language.get(keyword)
-        if self.indent == 0 and not self.first:
-            self.result += "\n\n"
-        if self.first:
-            self.first = False
         if issubclass(new_grammar.context, KeywordSyntax):
-            self.indent += 1
+            new_target_indent = self.context.cur_indent + 1
             self.grammar = Grammar(
                 new_grammar.language(),
                 new_grammar.context(
-                    keyword, self.indent, self.context, self.snakefile, accepts_py
+                    keyword,
+                    new_target_indent,
+                    self.context,
+                    self.snakefile,
+                    accepts_py,
                 ),
             )
             self.context_stack.append(self.grammar)
             self.process_keyword_context()
-            return None
+
+            status = self.context.get_next_queriable(self.snakefile)
+            self.buffer += status.buffer
+            return status
 
         elif issubclass(new_grammar.context, ParameterSyntax):
             param_context = new_grammar.context(
-                keyword, self.indent + 1, self.language, self.snakefile
+                keyword, self.target_indent + 1, self.language, self.snakefile
             )
             self.process_keyword_param(param_context)
             self.context.add_processed_keyword(status.token)
-            return KeywordSyntax.Status(
+            return Syntax.Status(
                 param_context.token,
                 param_context.cur_indent,
                 status.buffer,
                 param_context.eof,
             )
```

Comment:
true, but this should never be a problem the user should see- i don't see an error to throw here unless saying 'Programmer error'

Reply:
it just needs to be correctly coded

Reply:
What do you mean "it just needs to be coded correctly"?

Reply:
i mean, there are only two cases when a keyword is recognised, each case is covered in the if clauses. so what would go in an `else`? it's not like any user input can cause an error here, only a programming error?

Reply:
Ok, given these are the only two supported keywords it would just be something like 
```py
else:
    raise UnsupportedSyntax()
```
or something. There's no need to add a test or anything and it will never end up getting raised to the user if we only have two types of Syntax. It just means that if we add another Syntax then we know we need to allow for the possibility of it in this `if` statement. Does that make sense?  
If you don't like the idea of it that's ok we can just leave it as is. It's just the Rust compiler requires branches to be exhaustive and it has re-programmed me 😅 so it's totally fine to ignore what I am saying.

Reply:
OK, I'm sold ;-O
perhaps i need to become a rustacean too...